client/admin/media/index.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import Gallery from 'react-grid-gallery';
+import { useListController, useQueryWithStore } from 'react-admin';
+import { useRef } from 'react';
+import { HOST } from '../App';
+
+const MediaGallery = props => {
+  const imagesInObject = useListController(props).data;
+  const fileInputRef = useRef(null);
+  const [file, setFile] = useState(null);
+  const imagesRaw = Object.values(imagesInObject);
+  const handleClick = event => {
+    fileInputRef.current.click();
+  };
+
+  const images = imagesRaw.map(image => ({
+    thumbnailWidth: 240,
+    thumbnailHeight: 240,
+    src: `${HOST}${image.relativeSrc}`,
+    thumbnail: `${HOST}${image.relativeSrc}`,
+  }));
+
+  const handleFileUpload = event => {
+    if (event.target.files[0]) {
+      setFile(event.target.files[0]);
+    }
+  };
+
+  return (
+    <div>
+      <button onClick={handleClick}>hello world</button>
+      <input
+        type="file"
+        onChange={handleFileUpload}
```

Comment:
is it possible to reuse what react admin form and media input provide for easier actions handling + existing data provider logic might need less changes?